src/Common/src/TypeSystem/Common/CastingHelper.cs

```diff
@@ -357,7 +357,14 @@ private static bool CanCastToClass(this TypeDesc thisType, TypeDesc otherType, S
                     if (curType == otherType)
                         return true;
 
-                    curType = curType.BaseType;
+                    if (curType.IsInterface)
+                    {
+                        curType = curType.Context.GetWellKnownType(WellKnownType.Object);
```

Comment:
The other implementation of casting logic implement this case as one-off check before this loop, for example: https://github.com/dotnet/corert/blob/master/src/Runtime.Base/src/System/Runtime/TypeCast.cs#L633

It is a bit more efficient because of you need this check only first time through the loop. It would be nice to do it same way here.